nautobot/docs/installation/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,405 @@
+# Region and Site to Location Migration Guide
+
+In Nautobot 2.0.0, all the `Region` and `Site` related data models are being migrated to use `Location`. Below is a comprehensive guide for Nautobot users and Nautobot App developers to migrate their `Region` and `Site` related models and apps to `Location`.
+
+## Migrate ObjectPermission instances in Nautobot from Region and Site to Location
+
+Nautobot admins need to migrate `Site` and `Reigon` Related `ObjectPermission` instances to `Location`. The correct way to do it are documented below with practical examples.
+
+### Region Specific ObjectPermission
+
+Add `Location` ContentType to `object_types` field of the `ObjectPermission` if it is not already present.
+
+Since `Location` has all the filters that `Region` had and they retain the same functionalities. We do not need to modify the `constraints` field of any `Region` specific `ObjectPermission` instances.
```

Comment:
Might be good to talk about adding a `location_type` constraint to the permission, e.g. if the old permission was one that allowed for editing Regions but not editing Sites? Same comment on the next section as well.